streaming/src/main/scala/org/apache/spark/streaming/api/java/JavaPairDStream.scala

```diff
@@ -336,7 +336,8 @@ class JavaPairDStream[K, V](val dstream: DStream[(K, V)])(
    * However, it is applicable to only "invertible reduce functions".
    * Hash partitioning is used to generate the RDDs with Spark's default number of partitions.
    * @param reduceFunc associative reduce function
-   * @param invReduceFunc inverse function
+   * @param invReduceFunc inverse function; such that for all x, invertible y:
+   *                      `invReduceFunc(reduceFunc(x, y), y) = x`
```

Comment:
I'm afraid this one would be false as it implies commutativity. Try with string concatenation.